src/external/governance/GovernanceContract.sol

```diff
@@ -0,0 +1,339 @@
+        validAddress(newTeamMultisig)
+        validAmount(newTimelockPeriod)
+    {
+        // -> set COMMUNITY_MULTISIG_ROLE as admin of itself
+        _setRoleAdmin(COMMUNITY_MULTISIG_ROLE, COMMUNITY_MULTISIG_ROLE);
+        // -> set COMMUNITY_MULTISIG_ROLE as admin of DEFAULT_ADMIN_ROLE
+        _setRoleAdmin(DEFAULT_ADMIN_ROLE, COMMUNITY_MULTISIG_ROLE);
+
+        // Set up TEAM_MULTISIG_ROLE role structure:
+        // -> set COMMUNITY_MULTISIG_ROLE as admin of TEAM_MULTISIG_ROLE
+        _setRoleAdmin(TEAM_MULTISIG_ROLE, COMMUNITY_MULTISIG_ROLE);
+
+        // grant COMMUNITY_MULTISIG_ROLE to specified address
+        _grantRole(COMMUNITY_MULTISIG_ROLE, newCommunityMultisig);
+        // grant COMMUNITY_MULTISIG_ROLE to specified address
+        _grantRole(TEAM_MULTISIG_ROLE, newTeamMultisig);
+
+        timelockPeriod = newTimelockPeriod;
+    }
+
+    //--------------------------------------------------------------------------
+    // Getter Functions
+
+    /// @inheritdoc IGovernanceContract
+    function getBeaconTimelock(address beacon)
+        external
+        returns (Timelock memory)
+    {
+        return beaconTimelock[beacon];
+    }
+
+    //--------------------------------------------------------------------------
+    // FeeManager Functions
+
+    /// @inheritdoc IGovernanceContract
+    function getFeeManager() external view returns (address) {
+        return feeManager;
+    }
+
+    /// @inheritdoc IGovernanceContract
+    function setFeeManager(address newFeeManager)
+        external
+        onlyRole(COMMUNITY_MULTISIG_ROLE)
+        validAddress(newFeeManager)
+    {
+        feeManager = newFeeManager;
+    }
+
+    //--------------------------------------------------------------------------
+    // Beacon Functions
+    //---------------------------
+    // Upgrade
+
+    /// @inheritdoc IGovernanceContract
+    function upgradeBeaconWithTimelock(
+        address beacon,
+        address newImplementation,
+        uint newMinorVersion
+    )
+        external
+        onlyCommunityOrTeamMultisig
+        accessableBeacon(beacon)
+        validAddress(newImplementation)
+    {
+        beaconTimelock[beacon] = Timelock(
+            true,
+            block.timestamp + timelockPeriod,
+            newImplementation,
+            newMinorVersion
+        );
+
+        emit BeaconTimelockStarted(
+            beacon,
+            newImplementation,
+            newMinorVersion,
+            block.timestamp + timelockPeriod
+        );
+    }
+
+    /// @inheritdoc IGovernanceContract
+    function triggerUpgradeBeaconWithTimelock(address beacon)
+        external
+        onlyCommunityOrTeamMultisig
+        accessableBeacon(beacon)
+        upgradeProcessAlreadyStarted(beacon)
+        timelockPeriodExceeded(beacon)
+    {
+        //set timelock to inaktive
+        beaconTimelock[beacon].timelockActive = false;
+
+        //Upgrade beacon
+        IInverterBeacon(beacon).upgradeTo(
```

Comment:
Just for my own understanding. So this governance contract will be the owner of all the beacons? If so, then do we also need a way to transfer ownership from the beacons to another address in case that is needed in the future?

Reply:
Nvm, just saw further down below that you addresses this :)